pkg/operator/audit/audit.go

```diff
@@ -0,0 +1,104 @@
+package audit
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	kyaml "k8s.io/apimachinery/pkg/util/yaml"
+	auditv1beta1 "k8s.io/apiserver/pkg/apis/audit/v1beta1"
+
+	"github.com/openshift/cluster-openshift-apiserver-operator/pkg/operator/v311_00_assets"
+	libgoapiserver "github.com/openshift/library-go/pkg/operator/configobserver/apiserver"
+)
+
+const (
+	policyAsset = "v3.11.0/openshift-apiserver/audit-policies-cm.yaml"
+)
+
+func NewAuditPolicyPathGetter() (libgoapiserver.AuditPolicyPathGetterFunc, error) {
+	policies, err := readPolicyNamesFromResource()
+	if err != nil {
+		return nil, err
+	}
+
+	return func(profile string) (string, error) {
+		const (
+			auditPolicyFilePath = "/etc/kubernetes/static-pod-resources/configmaps/openshift-apiserver-audit-policies"
+		)
+
+		// we expect the keys for audit profile in bindata to be in lower case and
+		// have a '.yaml' suffix.
+		key := fmt.Sprintf("%s.yaml", strings.ToLower(profile))
+		_, exists := policies[key]
+		if !exists {
+			return "", fmt.Errorf("invalid audit profile - key=%s", key)
+		}
+
+		return fmt.Sprintf("%s/%s", auditPolicyFilePath, key), nil
+	}, nil
+}
+
+// GetDefaultPolicy returns the default audit policy.
+func GetDefaultPolicy() (*auditv1beta1.Policy, error) {
+	return getPolicyFromResource("default.yaml")
+}
+
+func readPolicyNamesFromResource() (map[string]struct{}, error) {
+	bytes, err := v311_00_assets.Asset(policyAsset)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load asset name=%s - %s", policyAsset, err)
+	}
+
+	rawJSON, err := kyaml.ToJSON(bytes)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert asset yaml to JSON name=%s - %s", policyAsset, err)
+	}
+
+	cm := corev1.ConfigMap{}
+	if err := json.Unmarshal(rawJSON, &cm); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal audit policy profile=%s - %s", policyAsset, err)
+	}
+
+	policies := map[string]struct{}{}
+	for key := range cm.Data {
+		policies[key] = struct{}{}
+	}
+
+	return policies, nil
+}
+
+func getPolicyFromResource(key string) (*auditv1beta1.Policy, error) {
```

Comment:
looks like this should be moved to the tests if GetDefaultPolicy is removed.

Reply:
done